### PR TITLE
Add back-compat overloads to prevent breaking change

### PR DESCRIPTION
**Pull request by agocke**
The changes here are just adding the back compat overloads, updating the *unshipped files, and sorting them.

### CODE REVIEW

File: src/Compilers/CSharp/Portable/PublicAPI.Unshipped.txt [hunk 1]
```diff
@@ -325,19 +320,26 @@ static Microsoft.CodeAnalysis.CSharp.SyntaxFactory.SwitchExpression(Microsoft.Co
...
+virtual Microsoft.CodeAnalysis.CSharp.CSharpSyntaxVisitor<TResult>.VisitBaseExpressionTypeClause(Microsoft.CodeAnalysis.CSharp.Syntax.BaseExpressionTypeClauseSyntax node) -> TResult
 virtual Microsoft.CodeAnalysis.CSharp.CSharpSyntaxVisitor<TResult>.VisitDiscardPattern(Microsoft.CodeAnalysis.CSharp.Syntax.DiscardPatternSyntax node) -> TResult
+virtual Microsoft.CodeAnalysis.CSharp.CSharpSyntaxVisitor<TResult>.VisitNullableDirectiveTrivia(Microsoft.CodeAnalysis.CSharp.Syntax.NullableDirectiveTriviaSyntax node) -> TResult
+virtual Microsoft.CodeAnalysis.CSharp.CSharpSyntaxVisitor<TResult>.VisitPositionalPatternClause(Microsoft.CodeAnalysis.CSharp.Syntax.PositionalPatternClauseSyntax node) -> TResult
 virtual Microsoft.CodeAnalysis.CSharp.CSharpSyntaxVisitor<TResult>.VisitPropertyPatternClause(Microsoft.CodeAnalysis.CSharp.Syntax.PropertyPatternClauseSyntax node) -> TResult
+virtual Microsoft.CodeAnalysis.CSharp.CSharpSyntaxVisitor<TResult>.VisitRangeExpression(Microsoft.CodeAnalysis.CSharp.Syntax.RangeExpressionSyntax node) -> TResult
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Portable/PublicAPI.Unshipped.txt]: It's not easy to tell at a glance which changes are just moving members around and which are really meaningful.

Would it be reasonable if in a separate PR we made it a build error to have PublicAPI.txt in an unsorted order to prevent this kind of stuff from coming up in diffs?

Comment (agocke) [reply src/Compilers/CSharp/Portable/PublicAPI.Unshipped.txt]: Lemme just split the sorting into a different commit.

Comment (agocke) [reply src/Compilers/CSharp/Portable/PublicAPI.Unshipped.txt]: Done
